pkg/virt-operator/resource/generate/rbac/operator.go

```diff
@@ -466,6 +467,16 @@ func NewOperatorRole(namespace string) *rbacv1.Role {
 				Resources: []string{
 					"secrets",
 				},
+				ResourceNames: []string{
+					components.KubeVirtCASecretName,
+					components.KubeVirtExportCASecretName,
+					components.VirtHandlerCertSecretName,
+					components.VirtHandlerServerCertSecretName,
+					components.VirtOperatorCertSecretName,
+					components.VirtApiCertSecretName,
+					components.VirtControllerCertSecretName,
+					components.VirtExportProxyCertSecretName,
```

Comment:
I now see an issue with that patch. 

```
# k describe kv kubevirt -n kubevirt
...
    Last Probe Time:       2023-03-23T09:22:34Z
    Last Transition Time:  2023-03-23T09:22:34Z
    Message:               An error occurred during deployment: secrets "kubevirt-export-ca" is forbidden: User "system:serviceaccount:kubevirt:kubevirt-operator" cannot delete resource "secrets" in API group "" in the namespace "kubevirt"
    Reason:                DeploymentFailed
    Status:                True
    Type:                  Degraded
```
I experimented with backporting it to `v0.54.0`. The secret `kubevirt-export-ca` has been introduced in a more recent version (probably `v0.58.0`). So I did the **downgrade** by applying the operator manifest v0.54.0, and it overwrote the RBAC causing a failure when deleting the secret. As I understand, now if a new secret is introduced in some future version, then it will cause the same failure when downgrading.

@kylealexlane, @rmohr, @mhenriks any thought on that?

Reply:
Just to note: with my test, the downgrade process eventually suceeded:
```
# k -n kubevirt wait kv kubevirt --for condition=Available
kubevirt.kubevirt.io/kubevirt condition met
```
However, the observed version in KubeVirt CR was still reported as `v0.58.0`:
```
# k describe kv kubevirt -n kubevirt
...
Observed Kube Virt Version:                   0.58.0-1505129.dv15sp5.
Operator Version:                             v0.54.0
...
Target Kube Virt Version:                     0.54.0-103.dv15sp4.1
```

Reply:
> I experimented with backporting it to v0.54.0. The secret kubevirt-export-ca has been introduced in a more recent version (probably v0.58.0). So I did the downgrade by applying the operator manifest v0.54.0, and it overwrote the RBAC causing a failure when deleting the secret. As I understand, now if a new secret is introduced in some future version, then it will cause the same failure when downgrading.

Hm, applying on an existing install the older operator manifests is a non-supported downgrade. In order to downgrade you have to keep the current operator and reference an older release in the KubeVirt CR. Downgrading the operator itself can lead to a stuck operator due to non-processable resources or due to missing permissions.

Reply:
Ah.. So this is an unsupported scenario... Now, thinking that probably would be beneficial to have this documented in the user-guide because it used to work in most cases.

Okay, then please ignore this issue :)

Reply:
> Ah.. So this is an unsupported scenario... Now, thinking that probably would be beneficial to have this documented in the user-guide because it used to work in most cases.

Yes I think there is some confusion around this area. We also have this discussion driven by @xpivarc: https://groups.google.com/d/msgid/kubevirt-dev/CAFTR0GVYPMYafxMvi8T7RzEax4cYDvwudAuGcxFcmpyzmDyPGw%40mail.gmail.com?utm_medium=email&utm_source=footer

Reply:
Yeah, that thread in the mailing list about operator updates. In fact, now I realize that I’ve been always just applying the operator yaml any time that I wanted to switch the version. Irrespective of whether it was an update or downgrade (never even thought that it needed to be taken into account). But thinking now, it makes sense.